share/genbuild.sh

```diff
@@ -31,6 +31,7 @@ if [ "${BITCOIN_GENBUILD_NO_GIT}" != "1" -a -e "$(which git 2>/dev/null)" -a "$(
     # if latest commit is tagged and not dirty, then override using the tag name
     RAWDESC=$(git describe --abbrev=0 2>/dev/null)
     if [ "$(git rev-parse HEAD)" = "$(git rev-list -1 $RAWDESC 2>/dev/null)" ]; then
+        git checkout .
```

Comment:
NACK it should definitely not mess with the files in the tree here, this will lose developer changes

Reply:
I run each line in genbuild.sh in ubuntu 18.04 console manually.
The return value :
git diff-index --quiet HEAD --

is 1 . Then "DESC=$RAWDESC" will not run. 


git status

On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   .gitattributes

... omit ...

	modified:   test/util/rpcauth-test.py

no changes added to commit (use "git add" and/or "git commit -a")

Reply:
So what does `git diff` say for all the modified files?

Reply:
A lots of things like :

diff --git a/build-aux/m4/ax_boost_filesystem.m4 b/build-aux/m4/ax_boost_filesystem.m4
old mode 100644
new mode 100755

I use the Gitian with VMware that installing clean Ubuntu 18.04.

Reply:
Please let us know exact steps to reproduce from a fresh vanilla Ubuntu